### PR TITLE
Engine

### PR DESCRIPTION
**Pull request by sepandhaghighi**
#### What does this implement/fix? Explain your changes.
- `engine` parameter added to `play` function
- `README.md` modified
- `__play_win` function renamed to `__play_winsound`
- `__play_mac` function renamed to `__play_afplay`
- `__play_linux` function renamed to `__play_alsa`
#### Local tests on OSs

- [x] macOS
    - [x] Sonoma
- [x] Windows
    - [x] Windows 11
- [ ] Linux
    - [ ] Ubuntu 20.04

### CODE REVIEW

File: nava/params.py [hunk 1]
```diff
@@ -8,11 +9,25 @@
...
+
+    >>> import nava
+    >>> engine = nava.Engine.ALSA
+    """
+
+    AUTO = "auto"
+    WINSOUND = "winsound"
+    ALSA = "alsa"
+    AFPLAY = "afplay"
+
```
Comment (sadrasabouri) [inline nava/params.py:24]: Now when user requests for an unavailable engine (an engine which is available in the `Engine` enum but not in the user's system) we raise the following error:
```
nava.errors.NavaBaseError: Sound can not play due to some issues.
```
It's better to handle this exception differently somewhere. We may want to handle this in another PR, but I am raising it here just for the record.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
I tested it in MacOS and it's working fine. Minor changes requested.

File: test/function_test.py [hunk 2]
```diff
@@ -27,6 +28,13 @@
 True
 >>> nava.params._play_threads_counter == 44
 True
+>>> sys_platform = sys.platform
+>>> if sys_platform == "win32":
+...     sound_id = nava.play(os.path.join("others", "test.wav"), async_mode=True, engine=nava.Engine.WINSOUND)
+... elif sys_platform == "darwin":
+...     sound_id = nava.play(os.path.join("others", "test.wav"), async_mode=True, engine=nava.Engine.AFPLAY)
+... else:
+...     sound_id = nava.play(os.path.join("others", "test.wav"), async_mode=True, engine=nava.Engine.ALSA)
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply test/function_test.py:37]: This section of tests is not about the `__play_auto` function. Here, I tried to test each engine manually.

File: README.md [hunk 3]
```diff
@@ -130,32 +130,45 @@ except NavaBaseError as e:
     print(str(e))
 ```
 
+### Engine
+
+⚠️ The `engine` parameter has a default value of `AUTO` 
+
+```python
+from nava import play, Engine
+sound_id = play("alarm.wav", async_mode=True, engine=Engine.AFPLAY)
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply README.md]: fixed in https://github.com/openscilab/nava/pull/50/commits/4d3178cb261e1163fb24cffae0b1ba4fcccbbbeb

File: nava/params.py (hunk 1 above)
Comment (sepandhaghighi) [reply nava/params.py:24]: We should talk about it later. At the moment, all of the engines only work on one OS. In the future, it's possible that we'll have an engine that works on more than one OS. So, I believe that it's not a bad idea to give this option to the user to choose any engine anywhere.

**Review by sadrasabouri — APPROVED** [review]

**Pull request merged**